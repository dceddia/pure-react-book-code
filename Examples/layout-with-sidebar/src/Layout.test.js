import React from 'react';
import Layout from './Layout';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders properly', () => {
  const wrapper = mount(<Layout/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders the sidebar', () => {
  const wrapper = mount(<Layout/>);
  wrapper.find('.content button').simulate('click');
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('hides the sidebar', () => {
  const wrapper = mount(<Layout/>);
  wrapper.find('.content button').simulate('click');
  wrapper.find('.sidebar button').simulate('click');
  expect(toJson(wrapper)).toMatchSnapshot();
});
