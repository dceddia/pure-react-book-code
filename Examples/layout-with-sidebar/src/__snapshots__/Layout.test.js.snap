// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides the sidebar 1`] = `
<Layout>
  <div
    className="layout"
  >
    <Content
      isSidebarVisible={false}
      onShowSidebar={[Function]}
    >
      <div
        className="content"
      >
        some content here
        <button
          onClick={[Function]}
        >
          Show
        </button>
      </div>
    </Content>
  </div>
</Layout>
`;

exports[`renders properly 1`] = `
<Layout>
  <div
    className="layout"
  >
    <Content
      isSidebarVisible={false}
      onShowSidebar={[Function]}
    >
      <div
        className="content"
      >
        some content here
        <button
          onClick={[Function]}
        >
          Show
        </button>
      </div>
    </Content>
  </div>
</Layout>
`;

exports[`renders the sidebar 1`] = `
<Layout>
  <div
    className="layout"
  >
    <Sidebar
      onHide={[Function]}
    >
      <div
        className="sidebar"
      >
        some sidebar content
        <button
          onClick={[Function]}
        >
          Hide
        </button>
      </div>
    </Sidebar>
    <Content
      isSidebarVisible={true}
      onShowSidebar={[Function]}
    >
      <div
        className="content"
      >
        some content here
      </div>
    </Content>
  </div>
</Layout>
`;
